### PR TITLE
Test groups API

### PR DESCRIPTION
**Pull request by APIYOJENNIFER**
Merging this PR (issue #3) adds functionality for;

-  testing the group creation API.
-  test retrieving of a list of groups.
- test group deletion i.e. successful deletion, when permission is denied, when the group doesn't exist.

### CODE REVIEW

File: tests/test_groups_api.py [hunk 1]
```diff
@@ -24,3 +24,17 @@ def test_create_group_api(api_client, authenticate_user_with_token):
     api_client.credentials()
 
     assert response.status_code == status.HTTP_201_CREATED
+
+
+@pytest.mark.django_db
+def test_get_groups_api(api_client, authenticate_user_with_token):
```
Comment (ThatcherK) [inline tests/test_groups_api.py]: I would rather the name suggests the functionality being tested as opposed to saying 'api' or the http method used. for example test_group_list

File: tests/test_groups_api.py [hunk 2]
```diff
@@ -24,3 +24,17 @@ def test_create_group_api(api_client, authenticate_user_with_token):
...
+    """Test GET method for retrieving groups"""
+    token = authenticate_user_with_token()
+
+    url = reverse("group")
+    api_client.credentials(HTTP_AUTHORIZATION='Token ' + token)
+    response = api_client.get(url)
+
+    api_client.credentials()
+
+    assert response.status_code == status.HTTP_200_OK
```
Comment (ThatcherK) [inline tests/test_groups_api.py:47]: Also an another assertion of perhaps the length of the list returned.

**Review by ThatcherK — CHANGES_REQUESTED** [review]

**Review by wakibi — COMMENTED** [review]
Read about code coverage vs test coverage. Also, [this](https://django-testing-docs.readthedocs.io/en/latest/coverage.html) maybe a good place to start implementing; make sure the test coverage is about 70% or more

**Review by wakibi — COMMENTED** [review]
One more addition, add a CI pipeline (using github actions) to have these tests run on every code pushed.

**Review by wakibi — APPROVED** [review]

**Comment by APIYOJENNIFER** [comment]
@ThatcherK please review the changes requested when you get time

**Comment by ThatcherK** [comment]
@APIYOJENNIFER rebase

**Comment by APIYOJENNIFER** [comment]
> @APIYOJENNIFER rebase

Done

**Comment by APIYOJENNIFER** [comment]
@ThatcherK I'm done with rebasing

**Review by ThatcherK — APPROVED** [review]

**Pull request merged**